File: Strings/14. Longest Common Prefix/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        min_length = float("inf")
        for s in strs:
            min_length = min(min_length, len(s))
        
        for i in range(min_length):
            current_char = strs[0][i]
            for s in strs:
                if s[i] != current_char:
                    return s[:i]
        
        return strs[0][:min_length]

# Time: O(n * m)
# Space: O(1)

File: Strings/14. Longest Common Prefix/14. Longest Common Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  let min_length = Infinity;
  for (let s of strs) min_length = Math.min(min_length, s.length);

  for (let i = 0; i < min_length; i++) {
    const current_char = strs[0][i];
    for (let s of strs) {
      if (s[i] !== current_char) return s.slice(0, i);
    }
  }

  return strs[0].slice(0, min_length);
};

// Time: O(n * m)
// Space: O(1)
